### PR TITLE
CHORE: Added model and field valdators, customised admin dashboard

### PR DESCRIPTION
**Pull request by thenoblet**
- Created `validators.py` to centralise custom validation logic.
- Implemented model field validation for `User`, `UserProfile`, `Organisation` and `Role` models.
- Modified `models.py` to integrate validation with `clean()` methods.
- Customised the admin dashboard in `admin.py` for improved usability.
- Added a migration (`0002_alter_organisation_table.py`) to update the `Organisation` table schema.

These changes enhance data integrity and improve the user experience in the admin interface.

### CODE REVIEW

File: src/user_service/models.py [hunk 1]
```diff
@@ -248,14 +292,15 @@ class Organisation(models.Model):
...
+        validators=[OrgModelValidator.owner_validator]
     )
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
 
     class Meta:
         """Metadata for the organisations model."""
 
-        db_table = "organisations"
+        db_table = "organisation"
```
Comment (nanafox) [inline src/user_service/models.py]: Shouldn't table names be plural?

File: src/user_service/validators.py [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+        Raises:
+            ValidationError:
+                - If the owner field is not defined in the Organisation model.
+                - If the owner is not an instance of the related User model.
+        """
+        from .models import Organisation
+
+        try:
+            owner_field = Organisation._meta.get_field("owner")
+        except FieldDoesNotExist:
```
Comment (nanafox) [inline src/user_service/validators.py]: When you catch an exception and raise another one, you should be explicit about it. You can for example do something like
```python
try:
    something_that_may_raise_an_exception
except ThOriginalException as e:
    raise SomeOtherExceptionInstead from e

File: src/user_service/models.py [hunk 3]
```diff
@@ -157,31 +166,56 @@ class UserStation(models.TextChoices):
...
+        "Role",
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name="user_role",
+        validators=[UserModelValidator.role_validator]
     )
     organisation = models.OneToOneField(
-        "Organisation", on_delete=models.CASCADE, null=True, blank=True
+        "Organisation", on_delete=models.CASCADE, null=True, blank=True, related_name="user_org"
```
Comment (nanafox) [inline src/user_service/models.py:202]: The default related name Django gives is `organisation` which I think is enough. This makes it simpler to have something like
```python
user = User.objects.get(id=1)
user.organization  # insetad of having to do user.user_org
```

What do you think?

**Review by nanafox — CHANGES_REQUESTED** [review]
I've checked the additions and for the most part, it is good. Before we proceed to merge, how have they been tested to confirm they work? Do you have any screenshots you can share to verify that the validations and everything we have here works as intended?

**Comment by thenoblet** [comment]
> I've checked the additions and for the most part, it is good. Before we proceed to merge, how have they been tested to confirm they work? Do you have any screenshots you can share to verify that the validations and everything we have here works as intended?

Yep. Tested and works.

File: src/user_service/models.py (hunk 1 above)
Comment (thenoblet) [reply src/user_service/models.py]: Anything you say boss

File: src/user_service/validators.py (hunk 2 above)
Comment (thenoblet) [reply src/user_service/validators.py]: Well, exception chaining is good. Why not? Will help with logging... we can see the full chain of exceptions in the logs when debugging. 

I will make the change

File: src/user_service/models.py (hunk 3 above)
Comment (thenoblet) [reply src/user_service/models.py:202]: You'll have a problem with me on this project.... I write BRE you write AME. Massa

**Pull request merged**